gmso/core/topology.py

```diff
@@ -610,6 +631,59 @@ def update_topology(self):
         self.update_connection_types()
         self.is_typed(updated=True)
 
+    def delete_site(self, site):
+        """Remove `site` from the topology
+
+        This method takes a gmso.Site object `site` and
+        removes the site if it exists in the topology.
+
+        Parameters
+        ----------
+        site : gmso.Site
+            A Site object that exists on the topology.
+
+        Raises
+        -------
+        GMSOError
+            If `site` doesn't exist in the topology's sites.
+
+        Notes
+        -----
+        Removing a site from the topology removes any bond,
+        angles, dihedrals and impropers associated with the site.
+        """
+        if site not in self._sites:
+            raise GMSOError(f"Site {site} does not exist in the topology")
+        else:
+            self._sites.discard(site)
+            self._dead_sites.add(site)
+            for connection in site.connections:
+                if connection in self._connections:
```

Comment:
This, and making a half-dozen new internal variables, seems like a lot of overhead for this behavior. But I believe it carefully deals with our mess of pointers that we've gotten ourselves in and the only alternative I can propose is moving this behavior into the `Connection` class itself and letting it do the big if block there. Actually, I kinda like that idea, in case there is another need to delete sites from elsewhere.